_rules/SC4-1-2-role-has-required-states-and-properties.md

```diff
@@ -0,0 +1,131 @@
+
+### Expectation
+
+For each test target, the [required states and properties](https://www.w3.org/TR/wai-aria/#requiredState) for the role are present, unless the state or property has a default value listed under [implicit value for role](https://www.w3.org/TR/wai-aria-1.1/#implictValueForRole). The list of required states and properties for a role can be found in [WAI-ARIA 1.1](https://www.w3.org/TR/wai-aria)"
+
+**Note:** This rule does not test whether the required states and properties has a correct value, only that the attributes are present and [non-empty](#non-empty).
+
+## Assumptions
+
+- This rule assumes that there is no element that has an [implicit semantic role](#semantic-role), and at the same time does not define native attributes that are mapped to all of the [required states and properties](https://www.w3.org/TR/wai-aria/#requiredState) for this role. This assumption is the basis for limiting the Applicability to [explicit semantic roles](#semantic-role) only.
+
+## Accessibility Support
+
+This rule relies on browsers and assistive technologies to support leaving out [required states and properties](https://www.w3.org/TR/wai-aria-1.1/#requiredState) when an [implicit value for role](https://www.w3.org/TR/wai-aria-1.1/#implictValueForRole) is specified in [WAI-ARIA 1.1](https://www.w3.org/TR/wai-aria-1.1).
+
+**Note:** The required states and properties with implicit values can be found in the Core Accessibility API Mappings 1.1 [Overview of default values for missing required attributes](https://www.w3.org/TR/core-aam-1.1/#authorErrorDefaultValuesTable).
+
+## Background
+
+- [ARIA5: Using WAI-ARIA state and property attributes to expose the state of a user interface component](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/ARIA5)
+- [WAI-ARIA required states and properties](https://www.w3.org/TR/wai-aria-1.1/#requiredState)
+- [RFC 3986](https://www.ietf.org/rfc/rfc3986.txt)
+
+## Test Cases
+
+### Passed
+
+#### Passed example 1
+
+Element has required states (no properties required for this role)
+
+```html
+<div role="option" aria-selected="false"></div>
+```
+
+#### Passed example 2
+
+Implicit value for roles: ´option´ role has an implicit value (´false´) specified for the required state ´aria-selected´
+
+```html
+<div role="option"></div>
+```
+
+#### Passed example 3
+
+Implicit value for roles: ´option´ role has an implicit value (´false´) specified for the required state ´aria-selected´, here defined without value
+
+```html
+<div role="option" aria-selected></div>
+```
+
+#### Passed example 4
+
+Implicit value for roles: ´option´ role has an implicit value (´false´) specified for the required state ´aria-selected´, here defined with only whitespace instead of a value
+
+```html
+<div role="option" aria-selected=" "></div>
+```
+
+#### Passed example 5
+
+Element has required properties (no states required for this role)
```

Comment:
There is a required state (`aria-expanded`) but it has a default value, so this is a pass anyway. But the comment needs to be updated.